### PR TITLE
Add generated stackgres structs for SGDbOps and SGCluster

### PR DESCRIPTION
**Pull request by glrf**
## Summary

This PR adds the tools to auto generate go types from the stackgres CRDs

Used for: https://github.com/vshn/appcat-comp-functions/pull/23

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

File: apis/stackgres/v1/groupversion_info.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// +kubebuilder:object:generate=true
+// +groupName=vshn.appcat.vshn.io
```
Comment (Kidswiss) [inline apis/stackgres/v1/groupversion_info.go:2]: Shouldn't this be stackgres.io?

Or is it even necessary to have this comment here? We only need the deepcopy generation from the kubebuilder.

**Review by Kidswiss — APPROVED** [review]
Just a small nit, the rest LGTM.

File: apis/stackgres/v1/groupversion_info.go (hunk 1 above)
Comment (glrf) [reply apis/stackgres/v1/groupversion_info.go:2]: Ah yes of course. Copy pasta error

File: Makefile [hunk 2]
```diff
@@ -80,3 +80,21 @@ push-docker: build-docker ## Push docker image with the manager.
 
 $(golangci_bin): | $(go_bin)
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "$(go_bin)"
+
+
+.PHONY: generate-stackgres-crds
+generate-stackgres-crds:
+	curl https://gitlab.com/ongresinc/stackgres/-/raw/1.4.3/stackgres-k8s/src/common/src/main/resources/crds/SGDbOps.yaml?inline=false -o apis/stackgres/v1/sgdbops_crd.yaml
```
Comment (zugao) [inline Makefile:87]: maybe we should configure the version of the CRDs we download?

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**